design/pipeline-resources.md

```diff
@@ -0,0 +1,199 @@
+
+
+### Schema
+
+```yaml
+resources:        # types: pipelines | repositories | containers | packages
+  pipelines:
+  - pipeline: string  # identifier for the pipeline resource
+    type: enum  # type of the pipeline source like azurePipelines, Jenkins etc. 
+    connection: string  # service connection to connect to the source
+    source:  
+      name: string  # source defintion of the pipeline including the project i.e. projectName/definition
+      version: string  # version to pick the artifact, optional; defaults to Latest
+      branch: string  # branch to pick the artiafct, optional; defaults to master branch
+      tags: string # picks the artifacts on from the pipeline with given tag, optional; defaults to no tags.
+```
+
+The inputs inside `source` can change based on the pipeline type defined. The above schema is for the `type`: azurePipelines.
+
+
+### Examples
+
+If you need to consume another azurePipelines from the current project and you dont require setting branch, version and tags info, this can be shortened to:
+
+```yaml
+resources:         
+  pipelines:
+  - pipeline: SmartHotel      
+    type: azurePipelines
+    source: SmartHotel-CI  # name of the pipeline source definition
+```
+
+In case you need to consume an azurePipeline from other project, then you need to include the project name while providing source name.
+
+```yaml
+resources:         
+  pipelines:
+  - pipeline: SmartHotel      
+    type: azurePipelines
+    source: 
+      name: DevOpsProject/SmartHotel-CI  # name of the pipeline source from different project
+      branch: releases/M142
+```
+
+### `downloadArtifact` for pipelines
+
+Artifacts from the `pipeline` resource are automatically downloaded and made available for all the jobs. However, in any of the jobs, you can choose to override and download only specific artifacts using `downloadArtifact` shortcut. Once you use `downloadArtifact` to override and download a specific artifact, automatic artifact download behavior is removed and you need to specify all the artifacts you intend to download in the job.
+
+
+```yaml
+- job: deploy_windows_x86_agent
+  steps:
+  - downloadArtifact: SmartHotel   # pipeline resource identifier.
+    name: WebTier1  # artifact to download, optional; defaults to all the artifacts from the resource.
+    patterns: '**/*.zip'  # mini match pattern to download specific files, optional; defaults to all files.
+```
+
+Or to avoid downloading any of the artifacts at all:
+
+```yaml
+- downloadArtifact: none
+```
+
+
+Refer to [download artifacts](https://github.com/Microsoft/azure-pipelines-yaml/blob/master/design/pipeline-artifacts.md#downloading-artifacts-downloadartifact) for more details.
+
+Artifacts from the `pipeline` resource are downloaded to `$PIPELINES_RESOURCESDIR/<pipeline-identifier>/` folder.
```

Comment:
Can you clarify this? Isn't it something more like "Artifacts from the `pipeline` resource are downloaded into named folders in `$PIPELINES_RESOURCESDIR/<pipeline-identifier>`, e.g. `$PIPELINES_RESOURCESDIR/SmartHotel/default`?

Reply:
yes, suppose smartHotel pipeline publishes two artifacts 'Web' and 'DB'. There will be two artifacts downloaded to $PIPELINES_RESOURCESDIR/SmartHotel/ i.e. `$PIPELINES_RESOURCESDIR/SmartHotel/Web` and `$PIPELINES_RESOURCESDIR/SmartHotel/DB`.